typed/tofieldset.go

```diff
@@ -94,9 +94,31 @@ func (v *toFieldSetWalker) doScalar(t *schema.Scalar) ValidationErrors {
 }
 
 func (v *toFieldSetWalker) visitListItems(t *schema.List, list value.List) (errs ValidationErrors) {
+	// Keeps track of the PEs we've seen
+	seen := fieldpath.MakePathElementSet(list.Length())
+	// Keeps tracks of the PEs we've counted as duplicates
+	duplicates := fieldpath.MakePathElementSet(list.Length())
```

Comment:
I would've expected the implementation to use just one set and one loop; `if seen.Has(pe) { continue }`

Reply:
we're trying to add duplicates exactly once to the set. I need to know if I've already seen them to know if they are duplicates, and I need to know if I've already added them to the set.

Reply:
I guess Im proposing the following pseudocode: 

```go
seen := NewSet()
for item in list {
    if seen.Has(item) { continue }
    seen.Insert(item)
    listWithoutDuplicates.append(item)
}
```

Doesn't this do as you described?

Reply:
Yeah, I think that makes sense, thanks.

Reply:
can mark this as resolved, since after discussion offline it is not what you want